routing/index_graph_starter.cpp

```diff
@@ -19,162 +48,373 @@ Junction CalcProjectionToSegment(Segment const & segment, Junction const & junct
 
 namespace routing
 {
-// static
-Segment constexpr IndexGraphStarter::kStartFakeSegment;
-Segment constexpr IndexGraphStarter::kFinishFakeSegment;
+void IndexGraphStarter::AddFakeVertex(Segment const & existentSegment, Segment const & newSegment,
+                                      FakeVertex const & newVertex, bool isOutgoing,
+                                      bool isPartOfReal, Segment const & realSegment)
+{
+  auto const & segmentFrom = isOutgoing ? existentSegment : newSegment;
+  auto const & segmentTo = isOutgoing ? newSegment : existentSegment;
+  m_fake.m_outgoing[segmentFrom].insert(segmentTo);
+  m_fake.m_ingoing[segmentTo].insert(segmentFrom);
+  m_fake.m_segmentToVertex[newSegment] = newVertex;
+  if (isPartOfReal)
+  {
+    m_fake.m_realToFake[realSegment].insert(newSegment);
+    m_fake.m_fakeToReal[newSegment] = realSegment;
+  }
+}
+
+Segment IndexGraphStarter::GetSegment(FakeVertex const & vertex, uint32_t & newNumber) const
+{
+  for (auto const & v : m_fake.m_segmentToVertex)
```

Comment:
Please, convert this to a method of FakeGraph. Also, I propose you to extract FakeGraph into individual fake_graph.hpp/cpp.

Reply:
FakeVertex and FakeGraph are nested private classes of IndexGraphStarter. Do you propose to move it out or just to move implementation to separate file?

Reply:
I'm proposing to move it out.